src/native/corehost/corehost.cpp

```diff
@@ -84,12 +84,29 @@ bool is_exe_enabled_for_execution(pal::string_t* app_dll)
 #define CURHOST_EXE
 #endif
 
-void need_newer_framework_error()
+void need_newer_framework_error(const pal::string_t& dotnet_root)
 {
-    pal::string_t url = get_download_url();
-    trace::error(_X("  _ To run this application, you need to install a newer version of .NET Core."));
-    trace::error(_X(""));
-    trace::error(_X("  - %s&apphost_version=%s"), url.c_str(), _STRINGIFY(COMMON_HOST_PKG_VER));
+    pal::string_t host_path;
+    pal::get_own_executable_path(&host_path);
```

Comment:
Pass in `host_path`? We should already have it in places that use this, so we can avoid getting it again.